<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Identifying genetic trait/phenotype relevant cell type/state at single cell resolution • SCAVENGE</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="bootstrap-toc.css">
<script src="bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="pkgdown.css" rel="stylesheet">
<script src="pkgdown.js"></script><meta property="og:title" content="Identifying genetic trait/phenotype relevant cell type/state at single cell resolution">
<meta property="og:description" content="SCAVENGE (Single Cell Analysis of Variant Enrichment through 
  Network propagation of GEnomic data) optimizes the inference of functional 
  and genetic associations to specific cells at single-cell resolution.">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-home">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="index.html">SCAVENGE</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.0.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="articles/SCAVENGE.html">Get started</a>
</li>
<li>
  <a href="reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="articles/docker.html">Docker/Singularity Containers</a>
    </li>
  </ul>
</li>
<li>
  <a href="news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/sankaranlab/SCAVENGE/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="contents col-md-9">

<a href="https://github.com/sankaranlab/SCAVENGE" class="external-link"><img src="https://img.shields.io/badge/devel%20version-1.0.2-black.svg"></a> <a href="https://github.com/sankaranlab/SCAVENGE/actions" class="external-link"><img src="https://github.com/sankaranlab/SCAVENGE/workflows/rworkflows/badge.svg" alt="R build status"></a> <a href="https://github.com/sankaranlab/SCAVENGE/commits/master" class="external-link"><img src="https://img.shields.io/github/last-commit/sankaranlab/SCAVENGE.svg"></a> <a href="https://github.com/sankaranlab/SCAVENGE" class="external-link"><img src="https://img.shields.io/github/languages/code-size/sankaranlab/SCAVENGE.svg"></a> <a href="https://codecov.io/gh/sankaranlab/SCAVENGE" class="external-link"><img src="https://codecov.io/gh/sankaranlab/SCAVENGE/branch/master/graph/badge.svg"></a> <a href="https://cran.r-project.org/web/licenses/GPL%20(%3E=%202)" class="external-link"><img src="https://img.shields.io/badge/license-GPL%20(%3E=%202)-blue.svg" alt="License: GPL (&gt;= 2)"></a> ¶
<h4>
¶ Authors: <i>Fulong Yu</i> ¶
</h4>
<h4>
¶ README updated: <i>Feb-22-2023</i> ¶
</h4>
<!-- To modify Package/Title/Description/Authors fields, edit the DESCRIPTION file -->
<div class="section level2">
<h2 id="scavenge-identifying-genetic-traitphenotype-relevant-cell-typestate-at-single-cell-resolution">
<code>SCAVENGE</code>: Identifying genetic trait/phenotype relevant cell type/state at single cell resolution<a class="anchor" aria-label="anchor" href="#scavenge-identifying-genetic-traitphenotype-relevant-cell-typestate-at-single-cell-resolution"></a>
</h2>
<div class="section level3">
<h3 id="overview">Overview:<a class="anchor" aria-label="anchor" href="#overview"></a>
</h3>
<p>Co-localization approaches using genetic variants and single-cell epigenomic data are unfortunately uninformative for many cells given the extensive sparsity across single-cell profiles. Therefore, only a few cells from the truly relevant population demonstrate reliable phenotypic relevance. The global high-dimensional features of individual single cells are sufficient to represent the underlying cell identities or states, which enables the relationships among such cells to be readily inferred. By taking advantage of these attributes, SCAVENGE identifies the most phenotypically-enriched cells by co-localization and explores the transitive associations across the cell-to-cell network to assign each cell a probability representing the cell’s relevance to those phenotype-enriched cells via network propagation.</p>
<p>We developed a novel enrichment method (<strong>SCAVENGE</strong>) (Single Cell Analysis of Variant Enrichment through Network propagation of GEnomic data) that can discriminate between closely related cell types/states and score single cells for GWAS enrichment.</p>
<div align="center">
<p><img src="image/schematic-view_1.png" width="680" height="278"></p>
</div>
<p align="center">
Schematic view of SCAVENGE
</p>
<p>We’ve implemented <strong>SCAVENGE</strong> as an <code>R</code> package for computing single-cell based GWAS enrichments from fine-mapped posterior probabilities and quantitative epigenomic data (i.e. scATAC-seq and potentially other single-cell epigenome profiling methods).<br>
As single-cell genomic datasets grow in volume, we expect SCAVENGE will have great promise for efficiently uncovering relevant cell populations for more phenotypes or functions in different scenarios, which may expand beyond the complex trait genetic variants we have examined here. We welcome you to use SCAVENGE to discover more phenotype relevant cells!</p>
</div>
<div class="section level3">
<h3 id="installation">Installation:<a class="anchor" aria-label="anchor" href="#installation"></a>
</h3>
<p>Once all of the dependencies for <code>SCAVENGE</code> are installed, the package can be installed directly from GitHub by typing the following into an <code>R</code> console:</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>(<span class="sc">!</span><span class="fu">require</span>(<span class="st">"remotes"</span>)) <span class="fu">install.packages</span>(<span class="st">"remotes"</span>)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a>remotes<span class="sc">::</span><span class="fu">install_github</span>(<span class="st">"https://github.com/sankaranlab/SCAVENGE"</span>)</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(SCAVENGE)</span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="documentation">Documentation<a class="anchor" aria-label="anchor" href="#documentation"></a>
</h2>
<div class="section level3">
<h3 id="tutorial">Tutorial:<a class="anchor" aria-label="anchor" href="#tutorial"></a>
</h3>
<p>This web resource and vignette compiliation shows how to reproduce results of SCAVENGE analysis with monocyte count on a 10X PBMC dataset <a href="doc/SCAVENGE-vignette.pdf">[<strong>Vignette-pdf</strong>]</a>, <a href="doc/SCAVENGE-vignette.Rmd">[<strong>Vignette-R markdown code</strong>]</a>.</p>
<p>See the <a href="https://github.com/sankaranlab/SCAVENGE/wiki" class="external-link">[<strong>Wiki page</strong>]</a> for some information on preparing your GWAS data for SCAVENGE, including finemapping.</p>
</div>
<div class="section level3">
<h3 id="website">
<a href="https://sankaranlab.github.io/SCAVENGE" class="external-link">Website</a><a class="anchor" aria-label="anchor" href="#website"></a>
</h3>
</div>
<div class="section level3">
<h3 id="getting-started">
<a href="https://sankaranlab.github.io/SCAVENGE/articles/SCAVENGE" class="external-link">Getting started</a><a class="anchor" aria-label="anchor" href="#getting-started"></a>
</h3>
</div>
<div class="section level3">
<h3 id="faqs">FAQs:<a class="anchor" aria-label="anchor" href="#faqs"></a>
</h3>
<ul>
<li>
<em>What input data are accepted for SCAVENGE analysis?</em><br>
A: The count matrix of scATAC-seq data and fine-mapped variants from GWAS summary statistics (we provided a tutorial for fine-mapping analysis from GWAS <a href="https://github.com/sankaranlab/SCAVENGE/wiki" class="external-link">[<strong>Wiki page</strong>]</a>). Theoretically, GWAS summary statistics can be used as input but we do not recommend it because LD can obscure causal cell type identification.</li>
<li>
<em>Can I use scRNA-seq instead of scATAC-seq?</em><br>
A: It is not feasible for SCAVENGE analysis from scRNA-seq currently. We are actively developing this tool to be scalable to scRNA-seq, please stay tuned.</li>
<li>
<em>How can I request new feature?</em><br>
A: We open <a href="https://github.com/sankaranlab/SCAVENGE/discussions" class="external-link">[<strong>Discussions</strong>]</a> page, please feel free to discuss and post your ideas.</li>
</ul>
</div>
<div class="section level3">
<h3 id="citation">Citation:<a class="anchor" aria-label="anchor" href="#citation"></a>
</h3>
<p>If you used or adapted SCAVENGE in your study, please cite our paper <a href="https://www.nature.com/articles/s41587-022-01341-y" class="external-link">[<em>Nat Biotechnol</em>]</a> || <a href="https://pubmed.ncbi.nlm.nih.gov/35668323/" class="external-link">[<em>PubMed</em>]</a>.<br><em>Variant to function mapping at single-cell resolution through network propagation.</em></p>
</div>
<div class="section level3">
<h3 id="contact">Contact:<a class="anchor" aria-label="anchor" href="#contact"></a>
</h3>
<p>If you run into issues and would like to report them, you can use the “Issues” tab on the left hand side.<br>
Alternatively, you can contact authors: fyu{at}broadinstitute.org, lcato{at}broadinstitute.org and/or sankaran{at}broadinstitute.org.</p>
</div>
</div>

  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <div class="links">
<h2 data-toc-skip>Links</h2>
<ul class="list-unstyled">
<li><a href="https://github.com/sankaranlab/SCAVENGE/" class="external-link">Browse source code</a></li>
</ul>
</div>

<div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li>GPL (&gt;= 2)</li>
</ul>
</div>


<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing SCAVENGE</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>Fulong Yu <br><small class="roles"> Author, maintainer </small> <a href="https://orcid.org/0000-0002-6100-8300" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
</ul>
</div>



  </div>
</div>


      <footer><div class="copyright">
  <p></p>
<p>Developed by Fulong Yu.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
